eclass/toolchain-funcs.eclass

```diff
@@ -421,6 +421,49 @@ tc-has-openmp() {
 	return ${ret}
 }
 
+# @FUNCTION: tc-ensure-openmp
+# @DESCRIPTION:
+# If 'openmp' is in IUSE and enabled OR if 'openmp' is not in IUSE, test
+# whether the currently active compiler supports OpenMP and die if it does
+# not. If it dies, it gives a detailed error message to the user, warning
+# them of the failure. This function should be used in pkg_setup (and
+# optionally in pkg_pretend).
+tc-ensure-openmp() {
+	case "${EAPI:-0}" in
+		[0-5])
+			die "Unsupported EAPI=${EAPI:-0} (too old) for tc-ensure-openmp"
+			;;
+		*)
+			;;
+	esac
+
+	if [[ ${MERGE_TYPE} != binary ]]; then
+		local openmp_requested
+		local die_message
+		if in_iuse openmp && use openmp; then
+			die_message="Requested USE=\"openmp\", but compiler does not support OpenMP"
+			openmp_requested=1
+		elif ! in_iuse openmp; then
+			die_message="Active compiler does not have required support for OpenMP"
+			openmp_requested=1
+		fi
+
+		if [[ $openmp_requested ]] && ! tc-has-openmp; then
```

Comment:
So I think this would be more readable without the extra variable, as:

    if { ! in_iuse openmp || use openmp; } && ! tc-has-openmp; then

(note: you need to check the operator precedence).

Having a second in_iuse check at the end is a minor problem.